tests_and_analysis/test/euphonic_test/test_spectrum2dcollection.py

```diff
@@ -0,0 +1,214 @@
+    return get_spectrum2dcollection("quartz_fuzzy_map.json")
+
+
+@pytest.fixture
+def quartz_fuzzy_items() -> list[Spectrum2D]:
+    """Individual spectra corresponding to quartz_fuzzy_collection"""
+    return [get_spectrum2d(f"quartz_fuzzy_map_{i}.json") for i in range(3)]
+
+@pytest.fixture
+def inconsistent_x_item() -> Spectrum2D:
+    """Spectrum with different x values"""
+    item = get_spectrum2d("quartz_fuzzy_map_0.json")
+    item._x_data *= 2.
+    return item
+
+@pytest.fixture
+def inconsistent_x_units_item():
+    """Spectrum with different x units"""
+    item = get_spectrum2d("quartz_fuzzy_map_0.json")
+    item.x_data_unit = "1/bohr"
+    return item
+
+@pytest.fixture
+def inconsistent_x_length_item():
+    """Spectrum with different number of x values"""
+    item = get_spectrum2d("quartz_fuzzy_map_0.json")
+    item.x_data = item.x_data[:-2]
+    item.z_data = item.z_data[:-2, :]
+    return item
+
+@pytest.fixture
+def inconsistent_y_item():
+    """Spectrum with different y values"""
+    item = get_spectrum2d("quartz_fuzzy_map_0.json")
+    item.y_data = item.y_data * 2.
+    return item
+
+def rand_spectrum2d(seed: int = 1,
+                    x_bins: Optional[Quantity] = None,
+                    y_bins: Optional[Quantity] = None,
+                    metadata: Optional[OneLineData] = None) -> Spectrum2D:
+    """Generate a Spectrum2D with random axis lengths, ranges, and metadata"""
+    rng = np.random.default_rng(seed=seed)
+
+    if x_bins is None:
+        x_bins = np.linspace(*sorted([rng.random(), rng.random()]),
+                             rng.integers(3, 10),
+                             ) * ureg("1 / angstrom")
+    if y_bins is None:
+        y_bins = np.linspace(*sorted([rng.random(), rng.random()]),
+                             rng.integers(3, 10)) * ureg("meV")
+    if metadata is None:
+        metadata = {"index": rng.integers(10),
+                    "value": rng.random(),
+                    "tag": "common"}
+
+    spectrum = Spectrum2D(x_data=x_bins,
+                          y_data=y_bins,
+                          z_data=rng.random([len(x_bins) - 1, len(y_bins) - 1]
+                                            ) * ureg("millibarn / meV"),
+                          metadata=metadata)
+    return spectrum
+
+
+class TestSpectrum2DCollectionCreation:
+    """Unit tests for Spectrum2DCollection constructors"""
+    def test_init_from_numbers(self):
+        """Construct Spectrum2DCollection with __init__()"""
+        n_x = 10
+        n_y = 20
+        n_z = 5
+
+        x_data = ureg.Quantity(np.linspace(0, 100, n_x), "1 / angstrom")
+        y_data = ureg.Quantity(np.linspace(0, 2000, n_y), "meV")
+        z_data = ureg.Quantity(np.random.random((n_z, n_x, n_y)), "1 / meV")
+
+        metadata = {"flavour": "chocolate",
+                    "line_data": [{"index": i} for i in range(n_z)]}
+
+        x_tick_labels = [(0, "Start"), (n_x - 1, "END")]
```

Comment:
Weird that "start" is title and "end" is upper.

Reply:
I think this was intended to catch any unexpected case-normalisation as a side-effect of logic.

Reply:
Makes sense!